### PR TITLE
[KernelGen][Nvidia] Add cudnn_batch_norm_backward operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary
Adds a Triton kernel for `cudnn_batch_norm_backward`. Computes gradients of input, weight, and bias for cuDNN style batch normalization backward pass.

## Testing
- Parametrized tests over shapes (2D-5D) and dtypes (float32, bfloat16)
- Validated against PyTorch reference on device side
- Tested on: Nvidia H20

## Performance
Test command: `pytest benchmark/test_cudnn_batch_norm_backward.py --level core` (NVIDIA H20)

| Configuration | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| (4, 16, 64, 4) | 0.008032 | 0.006816 | 1.178x |
| (16, 16, 8, 48) | 0.013088 | 0.008896 | 1.471x |
| (16, 16, 8, 88) | 0.018368 | 0.01136 | 1.617x |
| (16, 16, 128) | 0.00912 | 0.00784 | 1.163x |
| (20, 6, 65536) | 1.756 | 0.408864 | 4.295x |
| **Arithmetic Mean** | — | — | **1.945x** |

## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (5 cases, --level core) | 1.945x | Primary |
| Tianshu | FAIL | PASS (20 cases) | 1.240x | output_mask |
| Muxi | FAIL | FAIL (0 cases) | — | Operator not found on backend |
| Ascend | FAIL | N/A | — | aten::cudnn_batch not supported |
| Hygon | FAIL | FAIL (0 cases) | — | FakeScriptObject error |

## Files Changed
- `src/flag_gems/ops/cudnn_batch_norm_backward.py`: Triton kernel implementation
- `tests/test_cudnn_batch_norm_backward.py`: Accuracy test
- `benchmark/test_cudnn_batch_norm_backward.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: aten, stage: alpha 5.1)

### CODE REVIEW

**Comment by tengqm** [comment]
rebase your PR to master please, there is a conflict to solve.

File: benchmark/test_cudnn_batch_norm_backward.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import pytest
+import torch
+
+import flag_gems
+
+from . import base, consts
+
+
+# TODO(Qiming): Consolidate this to a base package
```
Comment (tengqm) [inline benchmark/test_cudnn_batch_norm_backward.py]: Remove this line

File: conf/operators.yaml [hunk 2]
```diff
@@ -903,6 +903,28 @@ ops:
       - Math
     stages:
       - stable: '2.2'
+  - id: cudnn_batch_norm_backward
```
Comment (tengqm) [inline conf/operators.yaml]: Please help make sure the list is sorted by operator ids.

File: benchmark/test_cudnn_batch_norm_backward.py (hunk 1 above)
Comment (bwbwzzz) [reply benchmark/test_cudnn_batch_norm_backward.py]: thanks，have revised

File: conf/operators.yaml (hunk 2 above)
Comment (bwbwzzz) [reply conf/operators.yaml]: thanks very， have revised

**Review by Caeruleann — APPROVED** [review]
